### PR TITLE
Simplify null checks

### PR DESCRIPTION
**Pull request by elachlan**
Refactoring to simplify null checks.

### CODE REVIEW

File: src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs [hunk 1]
```diff
@@ -1239,7 +1239,7 @@ private void PrintTargetNamePerMessage(BuildMessageEventArgs e, bool lightenText
                 string targetName = string.Empty;
 
                 // Does the context (Project, Node, Context, Target, NOT task) of the previous event match the current message
-                bool contextAreEqual = s_compareContextNodeIdTargetId.Equals(currentBuildEventContext, _lastDisplayedBuildEventContext == null ? null : _lastDisplayedBuildEventContext);
+                bool contextAreEqual = s_compareContextNodeIdTargetId.Equals(currentBuildEventContext, _lastDisplayedBuildEventContext ?? null);
```
Comment (BenVillalobos) [inline src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs:1242]: Is this even worth a null check? Even if not, I'd rather not make the change in this PR so we keep this one focused.

File: src/Tasks/GenerateResource.cs [hunk 2]
```diff
@@ -1833,7 +1833,7 @@ private bool NeedSeparateAppDomain()
                                             (
                                                 MessageImportance.Low,
                                                 "GenerateResource.SeparateAppDomainBecauseOfType",
-                                                (name == null) ? String.Empty : name,
+name ?? String.Empty,
```
Comment (BenVillalobos) [inline src/Tasks/GenerateResource.cs:1836]: Fix formatting here and in the other 2 changes below.

File: src/Tasks/GenerateResource.cs [hunk 3]
```diff
@@ -1879,7 +1879,7 @@ private bool NeedSeparateAppDomain()
                                             (
                                                 MessageImportance.Low,
                                                 "GenerateResource.SeparateAppDomainBecauseOfMimeType",
-                                                (name == null) ? String.Empty : name,
+name ?? String.Empty,
```
Comment (BenVillalobos) [inline src/Tasks/GenerateResource.cs:1882]: same as above

File: src/Tasks/GenerateResource.cs [hunk 4]
```diff
@@ -1925,7 +1925,7 @@ private bool NeedSeparateAppDomain()
                                         MessageImportance.Low,
                                         "GenerateResource.SeparateAppDomainBecauseOfErrorDeserializingLineNumber",
                                         source.ItemSpec,
-                                        (name == null) ? String.Empty : name,
+name ?? String.Empty,
```
Comment (BenVillalobos) [inline src/Tasks/GenerateResource.cs:1928]: same as above

File: src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs (hunk 1 above)
Comment (elachlan) [reply src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs]: There was a refactor suggestion to remove it. So I did. I think it comes under a simplification of null checks.

**Pull request merged**